File: node/wags-media-repository-web/src/models/Movie.ts
import { VideoGenre, VideoService } from "./System";

export type Movie = {
    movieId: number;
    title: string;
    imdbLink: string;
    posterImageUrl: string;
    dateWatched: Date;
    rating: number;
    thoughts: string;
    sortOrder?: number;
    statusId: number;
    genres: VideoGenre[];
    services: VideoService[]
}

File: tools/WagsMediaGenerator/Models/Movie.cs
namespace WagsMediaGenerator.Models;

public class Movie
{
    public int MovieId { get; set; }

    public string Title { get; set; } = string.Empty;

    public string ImdbLink { get; set; } = string.Empty;
    
    public DateTime? DateWatched { get; set; }
    
    public int? SortOrder { get; set; }
    
    public int Rating { get; set; }

    public string Thoughts { get; set; } = string.Empty;
    
    public string PosterImageUrl { get; set; } = string.Empty;

    public Tag Status { get; set; } = new();

    public IReadOnlyCollection<Tag> Genres { get; set; } = [];

    public IReadOnlyCollection<Tag> Services { get; set; } = [];
}